### PR TITLE
better event page: vimeo embed, event agenda

### PR DESCRIPTION
**Pull request by derekeder**
For events with Vimeo links, shows that as an embed at the top:

Screen Shot 2023-09-01 at 4 28 35 PM

Formats the agenda as a table with links to ordinances and sponsors:

Screen Shot 2023-09-01 at 4 31 21 PM

### CODE REVIEW

File: chicago/models.py [hunk 1]
```diff
@@ -172,6 +172,30 @@ def most_recent_past_city_council_meeting(cls):
         else:
             return None
 
+    @property
+    def clean_agenda_items(self):
```
Comment (derekeder) [inline chicago/models.py:176]: @fgregg I pulled this over from `django-councilmatic` and added a `prefetch_related` but I think I did it wrong. The page is very slow to load still

File: chicago/models.py [hunk 2]
```diff
@@ -172,6 +172,30 @@ def most_recent_past_city_council_meeting(cls):
...
+                descriptions_seen.append(a.description)
+                agenda_deduped.append(a)
+
+        return agenda_deduped
+
+    @property
+    def video_vimeo_id(self):
+        try:
+            link = self.media.first().links.first().url
+            vimeo_id = re.match(".*?([0-9]+)$", link).group(1)
```
Comment (derekeder) [inline chicago/models.py:193]: assuming the links are all vimeo and that the ID is always the last part of it, which seems to be the case

**Comment by fgregg** [comment]
optimized a lot of the queries here. probably fast enough now @derekeder

File: chicago/models.py [hunk 3]
```diff
@@ -172,6 +175,35 @@ def most_recent_past_city_council_meeting(cls):
...
+        )
+
+        return agenda_items
+
+    @property
+    def video_vimeo_id(self):
+        try:
+            link = self.media.first().links.first().url
+            vimeo_id = re.match(".*?([0-9]+)$", link).group(1)
+            print(link, vimeo_id)
```
Comment (fgregg) [inline chicago/models.py:202]: remove this print statement.

**Review by fgregg — CHANGES_REQUESTED** [review]

**Pull request merged**